.github/workflows/semantic.yaml

```diff
@@ -0,0 +1,35 @@
+name: "PR Lint"
+
+on:
+  pull_request_target:
+    types:
+      - opened
+      - edited
+      - synchronize
+
+jobs:
+  main:
+    name: Validate PR title
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out repository code
+        uses: actions/checkout@v3
+        with:
+          submodules: recursive
+      - uses: amannn/action-semantic-pull-request@v4
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          types: |
+            [feat]
+            [fix]
+            [docs]
```

Comment:
Cherry pick my comment from the gdoc proposal/document:

When working on https://github.com/apache/pulsar/pull/16836 you may find the standard https://www.conventionalcommits.org/ uses form:

> type(component):

What if we just follow the existing standard instead of creating a new one? That will satisfy all contributors already familiar to conventional commits.

Reply:
You may notice that you have to write `[feat]:` instead of `[feat]` in the title, that's how the conventional commits standard works.

Reply:
cc @Anonymitaet @lhotari @dave2wave @codelipenghui @eolivelli as you participant the [original discussion](https://lists.apache.org/thread/sk9ops3t94jmzc5tndk08y9khf7pj6so). I have some common concerns on this format also ><

Reply:
Also, if you follow conventional commits, there is already a [semantic-release](https://github.com/semantic-release/semantic-release) to general release note - we don't have to write and maintain our own variant.

Reply:
To be clear, I don't object to the PR naming conventions proposal. It's great to write down how we develop Apache Pulsar and enforce the rules. These comments are for the implementation parts and hope we reduce frictions to existing standard.

Reply:
Thanks your patient review~, I will draft this PR first, and reopen when all is ready.

Reply:
@tisonkun thanks! I think it makes sense to have our own standards. As you can see on the [PR list](https://github.com/apache/pulsar/pulls), more and more ppl stick to this standard `[xxx][xxx]xxxx`.

Reply:
@Anonymitaet It should trade off a random pick standard with significant efforts to build tools around it instead of following existing standards.

Before we apply a forced rule, we still have the last chance to change our decision. I'm -1 on this direction, but you have the binding vote :)

Reply:
About the percentage, when I count at the moment on https://github.com/apache/pulsar/commits/master (e1050e13f9e7c88da3a2ebdebbca7c5bdd52c1a1 to 5df15dd2edd7eeab309fea35828915c8698ea339)

People who follow exactly the rule = about 18
People who don't = 17

So It's not true that we can't change the status quo.

Reply:
The current standard is similar to the convention message, it should be a variant.  

If we use the convention message, we can focus on other things, and reduce time on these tools, but the developer gets used to the current standard. If we don't consider these, I suggest using the convention message.

Reply:
My concern is that enforcing a standard and not gently help users, especially new ones, come into compliance is simply adding barriers to our users. We should be prioritizing patterns that increase new users ability to contribute to the project. We must be willing to help users make the correct naming by changing it for them rather than make them do anything before their important (to them) contributions can be acted on.

Reply:
After an offline discussion with @Anonymitaet & @nodece we noticed that `action-semantic-pull-request` can define its own pattern to catch up on type, scope, and subject (`headerPattern`). So the technical part is not a blocker.

@dave2wave I agree that committers should help correct the title and help contributors to get used to the conventions. That's how I act as a committer in other projects.

BTW, enforcing such rules will significantly change contributors' experience so we should vote on the dev@ mailing list with an inline proposal (instead of a gdoc out of control to PMC).

The proposal doesn't need to be too long, but a brief motivation, the pattern proposed, implementation and existing workflows/conventions depend on it. (For example, RM can generate release note based on the pattern)

Reply:
Besides, currently we have some PR with title `[branch-xyz] ...` like https://github.com/apache/pulsar/pull/16914.

If we enforce any rule, this kind of example should be considered.

Reply:
Thanks! and @Anonymitaet has already mad a PIP about [Standardize PR Naming Convention using GitHub Actions](https://docs.google.com/document/d/1sJlUNAHnYAbvu9UtEgCrn_oVTnVc1M5nHC19x1bFab4/edit?pli=1), when the PIP passed, I will improve this PR:

1. (Optional) Implement semantic Github Action in [docbot](https://github.com/apache/pulsar-test-infra/tree/master/docbot)
2. Add a Job in [ci-documentbot.yml](https://github.com/apache/pulsar/blob/master/.github/workflows/ci-documentbot.yml) to check PR title and set `false` to `required`